### DISCUSSION TITLE
Using nuxt-edge in a package, then nuxt in a subdir/package yields error

### DISCUSSION THREAD

**Discussion by hecktarzuli** [nuxt/nuxt#7973] [General]
My docs are in the same repo as my module.  When trying to dev the docs I get this error.
If the module and docs have their own node_modules, package.json and yarn.lock etc.. why does this error out?

![image](https://user-images.githubusercontent.com/2112282/91383600-22319c80-e7f2-11ea-9814-0e4170ccb42e.png)

**Comment by debs-obrien** [comment]
yes i noticed that too. quick work around is remove nuxt-edge from the main package.json, work on docs. then don't commit the change to the package.json. not ideal but it works so at least you can move forward.

**Comment by hecktarzuli** [comment]
Ah, seems like it may be unexpected then.  Should I turn this into an issue?  
@Atinux @pi0

**Comment by pi0** [comment]
@hecktarzuli This check is intentional to avoid package version conflicts (see #6020 and #5996). You should use consistent distro (edge or non-edge) for both module and docs. Alternatively you can list `@nuxt/content-theme-docs` in top level package.json as devDependency so `yarn nuxt docs` will work. (package.json inside `docs` is to avoid installing unnecessary dependencies in docs deployment pipeline)

**Reply by hecktarzuli** [reply]
I'm curious what the significance of a package.json 2 levels down from the current one is?  It seems quasi-arbritrary, but I'm sure I'm missing something  :)

For example, it seems had my docs been in /something/docs instead of /docs/ I wouldn't get this error?

**Reply by pi0** [reply]
Actually yes it would! Thanks to `node_modules` searching algorithm when require a dependency, node will search all levels up to filesystem root for it so if any of parents have it installed we get the same error.

BTW agree it is not best DX even if there is a chance of dependency conflicts. For first version we was intentionally only checking one level ([here](https://github.com/nuxt/nuxt.js/pull/6020/files#diff-e4fda0311b18f50a7249c835b56ff3b4R11)) but with #6051, we switched to [packageExists](https://github.com/nuxt/nuxt.js/pull/6051/files#diff-e4fda0311b18f50a7249c835b56ff3b4R20) which checks all top levels. More thinking `npm:` alias was more edge case than `docs` requirement so maybe worth reverting to original version. /cc @clarkdo

**Comment by hecktarzuli** [comment]
per Pi0, since we want to catch problems early what seems to be the best setup is keeping nuxt-edge in the module, keeping nuxt:^### in the docs.  Then you just add, `@nuxt/content-theme-docs` to the root package.json as a dev dependency.  Then to dev the docs you just use `yarn nuxt docs` from the root.
